test/integration/deploy_trigger_test.go

```diff
@@ -192,14 +194,18 @@ waitForNewConfig:
 		case event := <-configWatch.ResultChan():
 			if event.Type == watchapi.Modified {
 				newConfig = event.Object.(*deployapi.DeploymentConfig)
-				break waitForNewConfig
+				// Multiple updates to the config can be expected (e.g. status
+				// updates), so wait for a significant update (e.g. version).
+				if newConfig.LatestVersion > 0 {
```

Comment:
The previous factoring also relied on watch or integration test timeouts, so it's no worse off. Makes sense to me to rely on the global timeout. Otherwise you have to choose arbitrary values or make config per test.